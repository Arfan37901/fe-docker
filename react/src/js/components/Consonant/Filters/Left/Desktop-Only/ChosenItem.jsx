import React from 'react';
import {
    string,
    func,
} from 'prop-types';

const ChosenFilterItemType = {
    id: string.isRequired,
    name: string.isRequired,
    onClick: func.isRequired,
    parentId: string.isRequired,
};

const ChosenFilterItem = (props) => {
    const {
        name,
        id,
        parentId,
        onClick,
    } = props;

    const handleClick = () => {
        onClick(parentId, id, false);
    };

    return (
        <button
            type="button"
            onClick={handleClick}
            data-testid="selected-filter"
            className="consonant-chosen-filter"
            tabIndex="0">
            {name}
        </button>
    );
};

ChosenFilterItem.propTypes = ChosenFilterItemType;

export default ChosenFilterItem;
