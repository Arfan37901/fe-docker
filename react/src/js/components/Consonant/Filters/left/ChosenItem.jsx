import React from 'react';
import PropTypes from 'prop-types';

const ChosenFilterItem = (props) => {
    const {
        name, id, parentId, onClick,
    } = props;

    const handleClick = () => {
        onClick(parentId, id, false);
    };

    return (
        <button
            type="button"
            onClick={handleClick}
            className="consonant-chosen-filter">{name}
        </button>
    );
};

export default ChosenFilterItem;

ChosenFilterItem.propTypes = {
    id: PropTypes.string.isRequired,
    parentId: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};
