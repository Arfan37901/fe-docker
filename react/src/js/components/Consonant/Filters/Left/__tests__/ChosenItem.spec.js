import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ChosenItem from '../Desktop-Only/ChosenItem';

import { DEFAULT_PROPS } from '../../../Helpers/Testing/Constants/SelectedFilter';

import makeSetup from '../../../Helpers/Testing/Utils/Settings';

const setup = makeSetup(ChosenItem, DEFAULT_PROPS);

describe('Consonant/ChosenItem', () => {
    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const { tree } = setup();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onOpen', () => {
            const {
                props: {
                    id,
                    name,
                    onClick,
                    parentId,
                },
            } = setup();

            const buttonElement = screen.getByText(name);

            fireEvent.click(buttonElement);

            expect(onClick).toBeCalled();
            expect(onClick).toHaveBeenCalledWith(parentId, id, false);
        });
    });
});
