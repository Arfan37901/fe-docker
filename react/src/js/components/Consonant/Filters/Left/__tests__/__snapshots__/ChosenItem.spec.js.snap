// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/ChosenItem Check snapshots should renders correctly 1`] = `
<button
  className="consonant-chosen-filter"
  data-testid="selected-filter"
  onClick={[Function]}
  type="button"
>
  filter name
</button>
`;
